api/controllers/EAppSubmittedController.js

```diff
@@ -0,0 +1,180 @@
+const sails = require('sails');
+const AWS = require('aws-sdk');
+const s3 = new AWS.S3();
+
+const inDevEnvironment = process.env.NODE_ENV === 'development';
+
+const EAppSubmittedController = {
+    async addDocsAndRenderPage(req, res) {
+        try {
+            const { uploadedFileData } = req.session.eApp;
+            if (uploadedFileData.length === 0) {
+                throw new Error('No files uploaded');
+            }
+            for (let i = 0; i <= uploadedFileData.length; i++) {
+                const endOfLoop = i === uploadedFileData.length;
+                if (endOfLoop) {
+                    return EAppSubmittedController._renderPageAndSendConfirmationEmail(
+                        req,
+                        res
+                    );
+                }
+                UploadedDocumentUrls.create(
+                    await EAppSubmittedController._dbColumnData(
+                        uploadedFileData[i],
+                        req
+                    )
+                )
+                    .then(() => {
+                        sails.log.info(
+                            `Url for document ${uploadedFileData[i].filename} added to db`
+                        );
+                    })
+                    .catch((err) => {
+                        throw new Error(err);
+                    });
+            }
+        } catch (err) {
+            sails.log.error(err.message);
+            res.serverError();
+        }
+    },
+
+    _renderPageAndSendConfirmationEmail(req, res) {
+        const queryParams = req.params.all();
+        const applicationId = queryParams.appReference;
+        const userDetails = {
+            firstName: req.session.account.first_name,
+            lastName: req.session.account.last_name,
+            email: req.session.email,
+            appType: req.session.appType,
+            userRef: req.session.user.id,
+        };
+
+        EAppSubmittedController._sendConfirmationEmail(
+            userDetails,
+            applicationId,
+            req
+        );
+
+        EAppSubmittedController._resetEAppSessionData(req);
+
+        return res.view('eApostilles/applicationSubmissionSuccessful.ejs', {
+            email: userDetails.email,
+            applicationId,
+            user_data: HelperService.getUserData(req, res), // needed for inner-header.ejs
+        });
+    },
+
+    _resetEAppSessionData(req) {
+        const newSessionData = {
+            s3FolderName: '',
+            uploadedFileData: [],
+            uploadMessages: {
+                errors: [],
+                infectedFiles: [],
+                fileCountError: false,
+            }
+        };
+        req.session.eApp = newSessionData;
+    },
+
+    _sendConfirmationEmail(userDetails, applicationId, req) {
+        const emailAddress = userDetails.email;
+        const applicationRef = applicationId;
+        const { domain } = req._sails.config.session;
```

Comment:
Take a look at how we do this in the notification service as we construct URLs for other emails and there is already a variable for this. It'll be better than working out the domain from the redis session.

https://github.com/UKForeignOffice/loi-notification-service/blob/develop/app/routes.js#L242

Reply:
Hey Simon, yeah the code looks a bit confusing but the domain is actually retrieved from the .env file but it happens to be part of the session object if that makes sense.

It looks like the notifications service is doing a similar thing but in a much more convoluted way. 
notify.js > common.js > server.js > routes.js

But that refactor is for a future PR :)

Reply:
Yes I see you are getting it from the redis session var, but now we have 1 email constructed differently than the rest of the emails in the service. 

Please change it to be constructed the same as the others in the notification service. It just makes it easier for future troubleshooting/development.

Reply:
This should be done now. Please let me know if you think anything else should be changed.